### ISSUE TITLE
Incorrect parquet dataPageOffset written

### ISSUE THREAD

**Issue by devinrsmith** [duckdb/duckdb#10829]
### What happens?

We have parquet reading logic that depends on `dataPageOffset` being correct - it appears this value is being written incorrectly with DuckDB generated parquet files for columns that have dictionaries. Other tools don't seem to be effected by this, likely because they read sequentially and don't need `dataPageOffset`.

This is what the page headers look like (generated using https://github.com/apache/parquet-mr/tree/master/parquet-cli):

```
Start of chunk (rowGroup: 0, columnName: X, dictPageOffset: 4, dataPageOffset: 142, numValues: 1001, totalSize: 344)
Page 0. (offset: 4, headerSize: 15)
{
  "compressed_page_size" : 138,
  "dictionary_page_header" : {
    "encoding" : 0,
    "num_values" : 32
  },
  "type" : 2,
  "uncompressed_page_size" : 182
}
Page 1. (offset: 157, headerSize: 20)
{
  "compressed_page_size" : 171,
  "data_page_header" : {
    "definition_level_encoding" : 3,
    "encoding" : 8,
    "num_values" : 1001,
    "repetition_level_encoding" : 3
  },
  "type" : 0,
  "uncompressed_page_size" : 2010
}
End of chunk (offset: 347)
```
The `dataPageOffset` for this chunk is 142. From what we understand, this should point to "Byte offset from beginning of file to first data page". We have two pages in this chunk, first is the dictionary page (notice the `dictionary_page_header`) and second is the data page (notice the `data_page_header`). The offset for page 1 or the first data page in the file is 157 and not 142, which I think is a mistake.

It looks like the code missed adding the `headerSize` from dictionary page to calculate the `dataPageOffset` for this chunk.

The same issue can be seen in all columns which use dictionary encoding.

Here's a link to our repository issue: https://github.com/deephaven/deephaven-core/issues/3651

### To Reproduce

```sql
COPY (
  SELECT
     CAST(generate_series % 32 AS STRING) as X 
  FROM
   generate_series(1000)
) TO 'bad-offsets.parquet' (FORMAT PARQUET);
```

### OS:

Linux x86_64 (Fedora 39)

### DuckDB Version:

0.10.0

### DuckDB Client:

CLI

### Full Name:

Devin Smith

### Affiliation:

Deephaven Data Labs

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Hi @devinrsmith, thanks for reporting this. Which Parquet CLI command did you use to generate the information on the page headers?

**Comment by devinrsmith** [comment]
I meant to ping my colleague @malhotrashivam, as he's the one who has done the deeper debugging. Here's the command he shared with me:

```sh
java -cp 'target/parquet-cli-1.14.0-SNAPSHOT.jar:target/dependency/*' org.apache.parquet.cli.Main pages --raw bad-offsets.parquet
```

**Comment by malhotrashivam** [comment]
Thanks Devin. Hi @szarnyasg, I built parquet-cli from source from this [github project](https://github.com/apache/parquet-mr/tree/master/parquet-cli) to generate the header information. The command was same as mentioned by Devin above:
```
java -cp 'target/parquet-cli-1.14.0-SNAPSHOT.jar:target/dependency/*' org.apache.parquet.cli.Main pages --raw bad-offsets.parquet
```
Let me know if you need any additional info related to parquet-cli. But it just dumps the headers. 

I also verified that the header information was correct by using our [deephaven-core](https://github.com/deephaven/deephaven-core) source code for reading parquet files. Although that requires a deeper dev understanding of our code.

**Comment by szarnyasg** [comment]
Thanks for the information! We'll take a look.

**Comment by Mytherin** [comment]
Fixed in #10890

**Comment by malhotrashivam** [comment]
Thanks for the prompt fix.
